### ISSUE TITLE
os x broken in ogles_gpgpu tests

### ISSUE THREAD

**Issue by headupinclouds** [headupinclouds/gatherer#124]
currently qmlvideofilter is not working for os x.  i've created a branch for fixes here:

https://github.com/headupinclouds/gatherer/tree/pr.develop.osx

in this branch i've made the default OGLESGPGPUTest pipeline a simple grayscale shader [here](https://github.com/headupinclouds/gatherer/blob/pr.develop.osx/src/lib/OGLESGPGPUTest/OGLESGPGPUTest.cpp#L116).  this ends up displaying a black texture.  

interestingly, if i send the input texture (created from BGRA camera input in the generic [MemTransfer::prepareInput](https://github.com/headupinclouds/ogles_gpgpu/blob/hunter/ogles_gpgpu/common/gl/memtransfer.cpp#L71) method) as the output of the VideoFilterRunnable, the texture seems to be created properly:

You can uncomment the line shown below ([here](https://github.com/headupinclouds/gatherer/blob/pr.develop.osx/src/app/qmlvideofilter/VideoFilterRunnable.cpp#L233) and the frame looks fine:

```
            //GLuint outputTexture = m_pipeline->getLastShaderOutputTexture();
            GLuint outputTexture = m_pipeline->getInputTexture();
```

When the  grayscale filter output is returned from  `getLastShaderOutputTexture` the display is black:

![black](https://cloud.githubusercontent.com/assets/554720/12705971/f67a4f06-c848-11e5-8351-dadfafb20c0a.jpg)

```
            GLuint outputTexture = m_pipeline->getLastShaderOutputTexture();
            //GLuint outputTexture = m_pipeline->getInputTexture();
```

But if you write to the bound output texture/fbo using the grayscale shader using some debugging code, then the debug pattern shows up fine in the display.  I.e., we still use the processed output texture

```
           GLuint outputTexture = m_pipeline->getLastShaderOutputTexture();
            //GLuint outputTexture = m_pipeline->getInputTexture();
```

but modify the grayscale filter to write hard coded values instead of texture/sampler dependent values from the input ([here](https://github.com/headupinclouds/ogles_gpgpu/blob/hunter/ogles_gpgpu/common/proc/grayscale.cpp#L38))

![rainbow](https://cloud.githubusercontent.com/assets/554720/12705981/13fb19de-c849-11e5-9cd0-aab8c779db2a.jpg)

```
    gl_FragColor = vec4(vTexCoord.x, vTexCoord.y, 0.5, 1.0);
```

**Comment by headupinclouds** [comment]
The same issue shows up the simple `glfw` based `ogles_gpgpu_test` sample, which I believe was working when I added initial OS X support to `ogles_gpgpu` a few months ago, so it might be possible to rewind to an earlier commit (when that was added) and try it there to see what has changed.  this could very well be a simple issue.  originally I was thinking it was due to an opengl profile incompatibility (`texture2D` calls from OpenGL ES 2.0 are deprecated in full more recent fledged OpenGL/GLSL versions), but then i noticed that the display render shader is working fine (which also uses `texture2D`).  maybe a missing `glActiveTeture` call?

**Comment by headupinclouds** [comment]
After rewinding to an earlier version I was able to sort through the OS X issues and the ogles_gpgpu_test app (glfw + ogles_gpgpu) runs fine.  The qmlvideofilter app is partially working on OS X.

https://github.com/headupinclouds/gatherer/tree/pr.develop.osx

**Comment by headupinclouds** [comment]
fixed in #125
